src/main/scala/mesosphere/marathon/core/appinfo/impl/DefaultInfoService.scala

```diff
@@ -31,6 +31,13 @@ private[appinfo] class DefaultInfoService(
       }
     }
 
+  override def selectPodStatuses(ids: Set[PathId], selector: PodSelector): Future[Seq[PodStatus]] = {
```

Comment:
I missed the opportunity to review initially, but I'd wondered why this needs to a `Set`, and then down below, I'd do `ids.iterator` followed by a `to[Seq]` to reduce the number of copies etc. Probably trivial optimization stuff :)

Reply:
The ids come as a set. I stumbled upon this as well.